src/api/uat/uat.ts

```diff
@@ -0,0 +1,32 @@
+import { QueryFunction, useQuery } from '@tanstack/react-query';
+import axios from 'axios';
+import { ADSQuery } from '../types';
+import { IATTermsSearchParams, IUATTermsSearchReponse } from './types';
+
+export enum UATSearchKeys {
+  TERM = 'term',
+}
+
+export const uatSearchKeys = {
+  term: (term: string) => [UATSearchKeys.TERM, term] as const,
+};
+
+export const useUATTermsSearch: ADSQuery<IATTermsSearchParams, IUATTermsSearchReponse> = (params, options) => {
+  return useQuery({
+    queryKey: uatSearchKeys.term(params.term),
+    queryFn: fetchUATTerms,
```

Comment:
I think we should set a cache time of around 30s so it doesn't fill up the cache for too long.  Probably the main case for caching is for a user who is editing a query (i.e. "s" -> "su" -> "sum" -> "su" -> "sun")

Reply:
👌